arch/arm/src/sama5/sam_flexcom_spi.c

```diff
@@ -0,0 +1,2116 @@
+        {
+          spierr("ERROR: Failed to allocate the RX DMA channel\n");
+          flex_spics->candma = false;
+        }
+    }
+
+  if (flex_spics->candma)
+    {
+      flex_spics->txdma = sam_dmachannel(flex_spino, 0);
+      if (!flex_spics->txdma)
+        {
+          spierr("ERROR: Failed to allocate the TX DMA channel\n");
+          sam_dmafree(flex_spics->rxdma);
+          flex_spics->rxdma  = NULL;
+          flex_spics->candma = false;
+        }
+    }
+#endif /* CONFIG_SAMA5_SPI_DMA */
+
+  /* Select the SPI operations */
+
+  /* have already used DEBUGASSERT to weed out invalid SPI port assignments */
+#ifdef CONFIG_SAMA5_FLEXCOM0_SPI
+  if (flex_spino == 0)
+    {
+      putreg32(FLEXCOM_MR_OPMODE_SPI, SAM_FLEX0_MR);
+      flex_spics->flex_spidev.ops = &g_flexcom0_spiops;
+      flex_spics->cs = flex_csno;
+      flex_spics->flex_spino = flex_spino;
+    }
+#endif
+
+#ifdef CONFIG_SAMA5_FLEXCOM1_SPI
+  if (flex_spino == 1)
+    {
+      putreg32(FLEXCOM_MR_OPMODE_SPI, SAM_FLEX1_MR);
+      flex_spics->flex_spidev.ops = &g_flexcom1_spiops;
+      flex_spics->cs = flex_csno;
+      flex_spics->flex_spino = flex_spino;
+    }
+#endif
+
+#ifdef CONFIG_SAMA5_FLEXCOM2_SPI
+  if (flex_spino == 2)
+    {
+      putreg32(FLEXCOM_MR_OPMODE_SPI, SAM_FLEX2_MR);
+      flex_spics->flex_spidev.ops = &g_flexcom2_spiops;
+      flex_spics->cs = flex_csno;
+      flex_spics->flex_spino = flex_spino;
+    }
+#endif
+
+#ifdef CONFIG_SAMA5_FLEXCOM3_SPI
+  if (flex_spino == 3)
+    {
+      putreg32(FLEXCOM_MR_OPMODE_SPI, SAM_FLEX3_MR);
+      flex_spics->flex_spidev.ops = &g_flexcom3_spiops;
+      flex_spics->cs = flex_csno;
+      flex_spics->flex_spino = flex_spino;
+    }
+#endif
+
+#ifdef CONFIG_SAMA5_FLEXCOM4_SPI
+  if (flex_spino == 4)
+    {
+      putreg32(FLEX_MR_OPMODE_SPI, SAM_FLEX4_MR);
+      flex_spics->flex_spidev.ops = &g_flexcom4_spiops;
+      flex_spics->cs = flex_csno;
+      flex_spics->flex_spino = flex_spino;
+    }
+#endif
+
+  /* Get the SPI device structure associated with the chip select */
+
+  flex_spi = flex_spi_dev(flex_spics);
+
+  /* Has the SPI hardware been initialized? */
+
+  if (!flex_spi->initialized)
+    {
+      /* Enable clocking to the flexcom SPI block(s) */
+
+      flags = enter_critical_section();
+#ifdef CONFIG_SAMA5_FLEXCOM0_SPI
+      if (flex_spino == 0)
+        {
+          sam_flexcom0_enableclk();
+
+          /* Configure multiplexed pins as connected on the board,
+           * by board-specific logic.
+           */
+        }
+#endif
+
+#ifdef CONFIG_SAMA5_FLEXCOM1_SPI
+      if (flex_spino == 1)
+        {
+          sam_flexcom1_enableclk();
+
+          /* Configure multiplexed pins as connected on the board,
+           * by board-specific logic.
+           */
+        }
+#endif
+
+#ifdef CONFIG_SAMA5_FLEXCOM2_SPI
+      if (flex_spino == 2)
+        {
+          sam_flexcom2_enableclk();
+
+          /* Configure multiplexed pins as connected on the board,
```

Comment:
just an option: switch-case

Reply:
as above, it's as per sam_spi.c so will double check and revise as appropriate

Reply:
@pkarashchenko same comment as earlier - is it actually neater as a switch?

Reply:
I was thinking about something like
```
#if defined(CONFIG_SAMA5_FLEXCOM0_SPI) || \
    defined(CONFIG_SAMA5_FLEXCOM1_SPI) || \
    defined(CONFIG_SAMA5_FLEXCOM2_SPI) || \
    defined(CONFIG_SAMA5_FLEXCOM3_SPI) || \
    defined(CONFIG_SAMA5_FLEXCOM4_SPI)
#  define SAMA5_FLEXCOM_SPINO
#endif
...
#ifdef SAMA5_FLEXCOM_SPINO
  uint8_t flex_spino;               /* SPI controller number (0 or 1) */
#endif
...
#ifdef SAMA5_FLEXCOM_SPINO
  switch (flex_spino)
    {
#ifdef CONFIG_SAMA5_FLEXCOM0_SPI
      case 0:
        sam_flexcom0_enableclk();
        break;
#endif
...
    }
#endif
```

Reply:
Aha! Much better idea and is then consistent with the flexcom usart code. Thanks :)